datasource/src/main/scala/quasar/physical/mongo/Tunnel.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.mongo
+
+import slamdata.Predef._
+
+import cats.effect._
+import cats.syntax.applicative._
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+import cats.syntax.apply._
+
+import com.jcraft.jsch._
+import com.mongodb.connection.ClusterSettings
+import com.mongodb.{ConnectionString, ServerAddress}
+
+import quasar.Disposable
+import quasar.concurrent.BlockingContext
+
+import scalaz.syntax.tag._
+
+import scala.collection.JavaConverters._
+
+object Tunnel {
```

Comment:
Maybe `Tunnel#settings` not `Settings#apply` then?

Reply:
I like `Settings.apply` better because this code does not necessarily involve tunnel stuff (because of lines 47/48), it also takes care of creating the `ClusterSettings` when there is no tunneling so it's a bit broader. If the entry point would have been `viaTunnel` then `Tunnel` would have been the better name.